xilem_web/src/events.rs

```diff
@@ -163,18 +162,25 @@ fn rebuild_event_listener<State, Action, V, Event>(
 ) where
     State: 'static,
     Action: 'static,
-    V: View<State, Action, ViewCtx, DynMessage>,
-    V::Element: ElementAsRef<web_sys::EventTarget>,
+    V: DomView<State, Action>,
     Event: JsCast + 'static + crate::Message,
 {
     ctx.with_id(ON_EVENT_VIEW_ID, |ctx| {
-        if prev_capture != capture || prev_passive != passive {
-            remove_event_listener(element.as_ref(), event, &state.callback, prev_capture);
+        element_view.rebuild(
+            prev_element_view,
+            &mut state.child_state,
+            ctx,
+            element.reborrow_mut(),
+        );
+        let was_created = element.flags.was_created();
+        if prev_capture != capture || prev_passive != passive || was_created {
+            if !was_created {
+                remove_event_listener(element.as_ref(), event, &state.callback, prev_capture);
+            }
```

Comment:
:thinking: I don't know why, but somehow i get a knot in my head when i read it
```rust
let was_created = // ..
if /*... */ || was_created {
   if !was_created {}
}
```

Reply:
Do you have alternative suggestions? I mean it doesn't look super beautiful, but does the job?

Reply:
Yes, it's definitely doing its job, and I only noticed it when my concentration was poor,
I don't want to waste your time here, sorry.

It's not significantly better, but it may still be a small thing: naming the condition + early return:

```rust
let was_created = element.flags.was_created();
let needs_update = prev_capture != capture || prev_passive != passive || was_created;
if !needs_update {
    return;
}
if !was_created {
    remove_event_listener(element.as_ref(), event, &state.callback, prev_capture);
}
state.callback =
    create_event_listener::<Event>(element.as_ref(), event, capture, passive, ctx);
```

Reply:
> I don't want to waste your time here, sorry.

All good, yeah your version is probably slightly more clear, thanks, I'll use it.